src/index.js

```diff
@@ -3,22 +3,35 @@ import ReactDOM from 'react-dom'
 import createHistory from 'history/createBrowserHistory'
 import AppContainer from 'react-hot-loader/lib/AppContainer'
 import App from './components/App'
+import loadMessages from './loadMessages'
 
 const history = createHistory()
+const LANG = 'en'
+const messages = {}
 
 const render = App =>
   ReactDOM.render(
     <AppContainer>
-      <App history={history} />
+      <App history={history} messages={messages} />
     </AppContainer>,
     document.getElementById('root')
   )
 
+function updateMessages(incoming) {
+  Object.assign(messages, incoming)
+  render(App)
+}
+
 if (process.env.NODE_ENV === 'development' && module.hot) {
   module.hot.accept('./components/App.js', () => {
     const App = require('./components/App').default
     render(App)
   })
+
+  module.hot.accept('./loadMessages', () => {
+    const loadMessages = require('./loadMessages').default
+    loadMessages(LANG).then(updateMessages)
+  })
```

Comment:
i dont know where my previous comment went. i guess it got removed with the latest changes.

but basically use require.resolveWeak without a promise server-side if you want to do it synchronously. are you saying something is wrong with `import()` when using the babel plugin without wrapping it in the `universal` hoc?